### PR TITLE
Add --skip-empty-emails and --default-email-domain flags to handle empty user emails

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes it so that when a user record has no email, the tool will by default log an error and exit. Two flags are also added to circumvent this:

- `--skip-empty-emails` - Simply ignores blank emails and leaves them blank in the output
- `--default-email-domain` - Used to generate an email address with the user's username and the provided email domain. Previously, this was hardcoded to `example.com`

Example:

```
mmetl transform slack --file export.zip --default-email-domain mydomain.com
```

For a user with the following slack json data:

```json
{
  "name": "myusername",
  "email": ""
}
```

results in the following output:

```json
{
  "username": "myusername",
  "email": "myusername@mydomain.com"
}
```

### CODE REVIEW

File: commands/check.go [hunk 1]
```diff
@@ -25,6 +25,9 @@ var CheckSlackCmd = &cobra.Command{
 func init() {
 	CheckSlackCmd.Flags().StringP("file", "f", "", "the Slack export file to transform")
 	CheckSlackCmd.Flags().Bool("debug", true, "Whether to show debug logs or not")
+	CheckSlackCmd.Flags().Bool("skip-email", false, "Ignore empty email addresses from the import file. Note that this results in invalid data.")
```
Comment (wiggin77) [inline commands/check.go:28]: What is the use case for ignoring empty emails and ending up with an invalid import?

File: commands/check.go [hunk 2]
```diff
@@ -25,6 +25,9 @@ var CheckSlackCmd = &cobra.Command{
...
+	CheckSlackCmd.Flags().String("email-domain", "", "If this flag is provided: When a user's email address is empty, the output's email address will be generated from their username and the provided domain.")
```
Comment (wiggin77) [inline commands/check.go:29]: Would it be cleared to call it `default-email-domain` ?

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   Couple of non-blocking questions.

File: commands/check.go (hunk 1 above)
Comment (mickmister) [reply commands/check.go:28]: @wiggin77 This is to help figure out which users failed to migrate. I suppose I was thinking a customer could grep through the output file and search for `"email":""` to see which users have the empty email address, though I suppose they could do that with the input too. I figured there should be an alternative that fits "I don't want the script to bail, and I don't have an email domain to auto-assign the users".

**Pull request merged**